File: eKnjiga/eKnjiga.Services/Database/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eKnjiga.Services.Database
{
    public class City
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;
        
        public int ZipCode { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Foreign key for Country
        public int CountryId { get; set; }
        [ForeignKey("CountryId")]
        public Country Country { get; set; }

        public ICollection<User> Users { get; set; }
    }
} 

File: eknjiga_mobile_app/lib/models/city.dart
class City {
  final int id;
  final String name;
  final int zipCode;

  City({required this.id, required this.name, required this.zipCode});

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      zipCode: json['zipCode'] as int? ?? 0,
    );
  }
}
